src/pages/index.js

```diff
@@ -6,121 +6,415 @@ import Section from "../components/Section.js";
 const cardUrlInput = document.querySelector("#card-url-input");
 
+const profileAvatarElement = document.querySelector(".profile__avatar");
+const profileAvatarUrlInput = document.querySelector("#avatar-url-input");
+const profileAvatarButton = document.querySelector(".profile__container-edit-icon");
+
 const editProfileModal = document.querySelector("#profile-edit-modal");
-const editProfileForm = editProfileModal.querySelector(".modal__form_profile");
+const editProfileForm = editProfileModal.querySelector("#edit-profile-form");
+const editProfileSubmitButton = editProfileModal.querySelector(".modal__button");
 
-const addNewCardModal = document.querySelector("#add-card-modal");
+//const updateAvatarModal = document.querySelector("#update-avatar-modal");
+const updateAvatarForm = document.querySelector("#update-avatar-form");
+// const updateProfileAvatarSubmitButton = updateAvatarModal.querySelector(".modal__button");
 
-const cardsListElement = document.querySelector(".cards__list");
+//const addNewCardModal = document.querySelector("#add-card-modal");
 const addCardForm = document.querySelector("#add-card-form");
+// const addNewCardSubmitButton = addNewCardModal.querySelector(".modal__button");
+
+
+
+const cardsListElement = document.querySelector(".cards__list");
+
+
+
 
 //CREATE NEW INSTANCES OF ALL CLASSES // INITIALIZE
 
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "7bed9883-2658-4b5e-974b-2c2e17824760",
+    "Content-Type": "application/json",
+  },
+});
+
+const deleteCardPopup = new PopupWithDeleteConfirmation(
+  "#delete-card-modal",
+  handleConfirmDeleteSubmit,
+);
+
+deleteCardPopup.setEventListeners();
+
 const cardPreviewPopup = new PopupWithImage("#preview-image-modal");
 //cardPreviewPopup.setEventListeners();
 
-const editProfileFormValidator = new FormValidator(config, editProfileForm);
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: createCard,
-  },
-  ".cards__list"
-);
-cardSection.renderItems();
+
+const editProfileFormValidator = new FormValidator(config, editProfileForm);
 
 const editProfilePopup = new PopupWithForm(
   selectors.editProfileModal,
   handleEditProfileFormSubmit
 );
-//editProfilePopup.setEventListeners();
-
-console.log(editProfileModal);
 
 const editUserInfo = new UserInfo({
   titleSelector: ".profile__title",
   occupationSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
 const addNewCardPopup = new PopupWithForm(
   selectors.addNewCardModal,
   handleAddCardFormSubmit
 );
-console.log(addNewCardModal);
+
+const updateAvatarPopup = new PopupWithForm(
+  selectors.updateAvatarModal, 
+  handleProfileUpdateAvatarFormSubmit
+);
 
 const addNewCardFormValidator = new FormValidator(config, addCardForm);
 addNewCardFormValidator.enableValidation();
 
+const updateAvatarFormValidator = new FormValidator(config, updateAvatarForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```